blocks/serializer/index.js

```diff
@@ -0,0 +1,38 @@
+/**
+ * External dependencies
+ */
+import * as query from 'hpq';
+
+/**
+ * Takes a block list and returns the serialized post content
+ *
+ * @param  {Array}  blocks Block list
+ * @return {String}        The post content
+ */
+export default function serialize( blocks ) {
+	return blocks.reduce( ( memo, block ) => {
+		const blockType = block.blockType;
+		const blockSettings = wp.blocks.getBlockSettings( blockType );
+		const rawContent = wp.element.renderToString(
+			blockSettings.save( block.attributes )
```

Comment:
I'm trying to be pragmatic here, we have several issues if we return an element (or a component) and I would like to move forward with this.

I think this resolves several issues:

- Imagine using a component that requires lifecycle hooks, the rendering would be async.
- How can we drop the `data-react` attributes easily
- We can not return multiple tags, A text block can have multiple paragraphs
- Currently the `content` parsed by the text element returns all the markup `<p>test</p>`, if we add a container element, we would change the markup even if we do not perform any change `<p><p>test</p></p>`.

The string return value resolves all these without tradeoffs IMO